File: src/main/java/com/battletrident/games/state/GameManager.java
package com.battletrident.games.state;

import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.entity.Player;

import static com.battletrident.Consts.getServer;

public class GameManager {
	private static final GameUpdateEvent gameState = new GameUpdateEvent(GameState.ENDED, null);

	public static void playGame() {
		gameState.setGameState(GameState.PLAYING, null);
		gameState.callEvent();
	}
	public static void playGame(Player executor) {
		gameState.setGameState(GameState.PLAYING, executor);
		gameState.callEvent();
	}
	public static void stopGame() {
		gameState.setGameState(GameState.ENDED, null);
		gameState.callEvent();
	}
	public static void stopGame(Player executor) {
		gameState.setGameState(GameState.ENDED, executor);
		gameState.callEvent();
	}

	public static boolean isPlaying() {
		return gameState.getGameState() == GameState.PLAYING;
	}
	public static boolean isEnded() {
		return gameState.getGameState() == GameState.ENDED;
	}
	public static boolean isMapChanging() {
		return gameState.getGameState() == GameState.MAP_CHANGING;
	}
}

File: src/main/kotlin/com/BattleTrident/games/state/GameManager.kt
package com.BattleTrident.games.state

import org.bukkit.entity.Player

class GameManager {
	val gameState = GameUpdateEvent(GameState.ENDED, null)

	val isPlaying: Boolean
		get() = gameState.getState() == GameState.PLAYING
	val isEnded: Boolean
		get() = gameState.getState() == GameState.ENDED
	val isMapChanging: Boolean
		get() = gameState.getState() == GameState.MAP_CHANGING

	fun start(executor: Player? = null) {
		gameState.setState(GameState.PLAYING, executor)
		gameState.callEvent()
	}
	fun stop(executor: Player? = null) {
		gameState.setState(GameState.ENDED, executor)
		gameState.callEvent()
	}
}